cmd/consul-dataplane/main.go

```diff
@@ -131,6 +133,8 @@ func init() {
 	StringVar(&readyBindAddr, "envoy-ready-bind-address", "", "DP_ENVOY_READY_BIND_ADDRESS", "The address on which Envoy's readiness probe is available.")
 	IntVar(&readyBindPort, "envoy-ready-bind-port", 0, "DP_ENVOY_READY_BIND_PORT", "The port on which Envoy's readiness probe is available.")
 	IntVar(&envoyConcurrency, "envoy-concurrency", 2, "DP_ENVOY_CONCURRENCY", "The number of worker threads that Envoy uses.")
+	IntVar(&envoyDrainTime, "envoy-drain-time", 30, "DP_ENVOY_DRAIN_TIME", "The time in seconds for which Envoy will drain connections.")
+	StringVar(&envoyDrainStrategy, "envoy-drain-strategy", "immediate", "DP_ENVOY_DRAIN_STRATEGY", "The behaviour of Envoy during the drain sequence. Determines whether all open connections should be encouraged to drain immediately or to increase the percentage gradually as the drain time elapses.")
```

Comment:
Set [reasonable defaults for service-to-service deployments](https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-drain-time-s), and allow user configuration if needed. This is net new functionality as the prior proxy manager logic would just immediately kill the Envoy process, so the stock defaults (10 minute drain time, `"gradual"` drain strategy) would never have had any effect.

Reply:
The only place I see `--drain-time-s` in non-test code in Consul is as an assumption that it's used to configure hot restarts, and to disallow such configuration (it can be used for hot restarts, but is also applicable to graceful shutdown behavior as implemented here).

Similar to https://github.com/hashicorp/consul-dataplane/pull/100#issuecomment-1542330523, this behavior appears to be net-new for k8s and does not have equivalent functionality in Consul core.